src/Faker/Provider/nl_NL/Person.php

```diff
@@ -346,4 +379,87 @@ public function idNumber()
         }
         return implode('', array_reverse($nr));
     }
+
+    /**
+     * @param string|null $documentCode Key from $idStartLetter
+     * @param array $date               'timestamp' of 'type' issue or expiry. 'birth' is birthdate of person
+     * @param string|null $timezone     Timezone to use
+     */
+    public function identityDocument(
+        $documentCode = null,
+        $date = [
```

Comment:
This is messy. It's not strict with an array here. Let's design a proper interface here

Reply:
I've refactored the dates to an abstract class (because of the need for a constructor. Please let me know where under which file/namespace you'd like to see those additional classes if they don't belong in `Person.php`.

Reply:
Abstract is not the way to go. What do you want to achieve with this?

Reply:
Idea of abstract instead of interface was because to avoid repeated constructor logic. But has been refactored anyway (no more issuedate & expirydate generation in Faker)